src/icalendar/parser.py

```diff
@@ -181,28 +183,40 @@ def q_join(lst, sep=",", always_quote=False):
     return sep.join(dquote(itm, always_quote=always_quote) for itm in lst)
 
 
-def single_string_parameter(func):
-    """Create a parameter getter/setter for a single string parameter."""
+def single_string_parameter(upper=False):
```

Comment:
Maybe define as `single_string_parameter(_func=None, *, upper=False):` instead and replace `return decorator` with

```python
if _func is None:
    return decorator
else:
    return decorator(_func)
```
That would allow for ommitting `()` in `@single_string_parameter()` and prevents an error, when `()` is missing.